Source/WebKit/NetworkProcess/storage/FileSystemStorageHandle.cpp

```diff
@@ -346,13 +345,64 @@ std::optional<FileSystemStorageError> FileSystemStorageHandle::executeCommandFor
     return FileSystemStorageError::Unknown;
 }
 
-std::optional<FileSystemStorageError> FileSystemStorageHandle::executeCommandForWritable(WebCore::FileSystemWritableFileStreamIdentifier streamIdentifier, WebCore::FileSystemWriteCommandType type, std::optional<uint64_t> position, std::optional<uint64_t> size, std::span<const uint8_t> dataBytes, bool hasDataError)
+void FileSystemStorageHandle::executeCommandForWritable(WebCore::FileSystemWritableFileStreamIdentifier streamIdentifier, WebCore::FileSystemWriteCommandType type, std::optional<uint64_t> position, std::optional<uint64_t> size, std::span<const uint8_t> dataBytes, bool hasDataError, CompletionHandler<void(std::optional<FileSystemStorageError>)>&& completionHandler)
 {
+    if (type == WebCore::FileSystemWriteCommandType::Write && !hasDataError) {
```

Comment:
I remember file size can be changed for [truncate()](https://fs.spec.whatwg.org/#api-filesystemwritablefilestream-truncate), so I think we should add quota check for that too.